File: Java/Medium/Accepted/array-nesting_6.java
class Solution {
    public int arrayNesting(int[] nums) {
        
        int n = nums.length;
        int max = 0;
        for (int i = 0; i < n; i++) {
            int count = 0;
            for (int j = i; nums[j] != -1; count++) {
                int temp = nums[j];
                nums[j] = -1;
                j = temp;
            }
            max = Math.max(max, count);
        }
        return max;
    }
}

File: Python3/Medium/Accepted/array-nesting_6.py
class Solution:
    def arrayNesting(self, nums: List[int]) -> int:
        
        visited = [False] * len(nums)
        max_len = 0
        for i in range(len(nums)):
            if not visited[i]:
                count = 0
                while not visited[i]:
                    visited[i] = True
                    i = nums[i]
                    count += 1
                max_len = max(max_len, count)
        return max_len